mahmoodhamdi/Blogify
# Add Blog Creation Page with Image Picker and Topic Selection

### Description
This PR introduces a new page for creating blog posts within the Blogify app. The `AddBlogPage` allows users to input a blog title, content, and select topics from a predefined list. Additionally, users can select a cover image for their blog post using the new image picker utility.

### Changes Made
- Added `AddBlogPage` with form fields for title, content, and topic selection.
- Integrated image picker functionality using the `pickImage` utility function.
- Implemented topic selection using chips for better user interaction.
- Added `TextConstants` for managing predefined topics and error messages.
- Updated `BlogPage` to include a button for navigating to the `AddBlogPage`.
- Created `BlogEditor` widget for enhanced text input experience.

### How to Test
1. Navigate to the Blog page.
2. Click on the add button to open the `AddBlogPage`.
3. Fill in the blog details, select an image, and choose topics.
4. Verify that the form works as expected and topics can be selected and deselected.
5. Ensure that the image picker allows selecting an image and displays it correctly on the page.

### Additional Notes
- The `pickImage` utility is currently set to pick images from the gallery. Future improvements could include adding camera functionality.
- The `GradientButton` in the form does not have a linked action yet, which will be implemented in future updates.
